lib/cylc/config.py

```diff
@@ -1450,9 +1443,28 @@ def generate_edges( self, lexpression, left_nodes, right, seq, suicide=False ):
             conditional = True
 
         for left in left_nodes:
-            if left is not None:
-                e = graphing.edge( left, right, seq, suicide, conditional )
-                self.edges.append(e)
+            if left is None:
+                continue
+            if right is not None:
+                # Check for self-suicide and self-edges.
+                if left == right or left.startswith(right + ':'):
+                    # (This passes inter-cycle offsets: left[-P1D] => left)
+                    # (TODO - but not explicit null offsets like [-P0D]!)
```

Comment:
I'm happy for this to be another issue, if you like. Intervals are compared by converting each one to days and seconds (and there is a get-a-null-interval classmethod), so it should be fine.